### ISSUE TITLE
I2C does not work on Pyboard 'D' with build v1.11-63

### ISSUE THREAD

**Issue by nherriot** [SamsungResearchUK-IoT-Meetup/multimode_sensor_platform#12]
When testing on Pyboard 'D' with compiled source at version:

**MicroPython v1.11-63-gd889def-dirty on 2019-07-02; PYBD-SF2W with STM32F722IEK**

I2C does not work for the HDC2080 tile sensor. To reproduce the bug on the pyboard do:

>>> import machine
>>> i2c = machine.I2C('X')
>>> i2c.scan() 
[]

Also if you try and read and write to an I2C device you will get OS Error 19.

>>> hdc = HDC2080(i2c)
>>> hdc.measure()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 11, in measure
OSError: [Errno 19] ENODEV

Tasks to fix are:

- [x] Compile latest version of micropython.org for F2 board with thread's
- [x] Compile latest version of micropython.org for F6 board with thread's
- [ ] Compile latest version of micropython.org for F3 board with thread's
- [x] Check I2C works with >>> i2c.scan() which should show 64 and 69
- [x] Check thread's work with >>> import _thread
- [x] Copy new firmware to the github this [repo](https://github.com/SamsungResearchUK-IoT-Meetup/multimode_sensor_platform/tree/master/firmware) for the F2, F4 and F6
- [x] Try running the new [HDC2080 sensor](https://github.com/SamsungResearchUK-IoT-Meetup/multimode_sensor_platform/blob/ISSU10-no-driver-for-tile/microserver/drivers/hdc2080_sensor.py) code on the new firmware

**Comment by nherriot** [comment]
I've compiled and pushed new firmware images for 2F2W and flashed to an SF2W board.
So far so good. 
This is in branch:
https://github.com/SamsungResearchUK-IoT-Meetup/multimode_sensor_platform/tree/ISSU13-I2C-does-not-work-on-PyboardD

I have not tested on SF3, SF6 yet.

Sensors seem to work intermittently. 
Once this is tested on SF3, SF6 we can close this off.

**Comment by nherriot** [comment]
Added a /test/test_sensor.py file which is based on the example code.
I've also added the config line to enable I2C on the new pyboard D.
Tested on SF2 and SF6. 
I don't have an SF3 - so this will have to be done by Christine I guess?

[test_sensors.py](https://github.com/SamsungResearchUK-IoT-Meetup/multimode_sensor_platform/tree/ISSU13-I2C-does-not-work-on-PyboardD/tests)

Now closing and merging this branch with master.
